eth_utils/conversions.py

```diff
@@ -115,13 +124,12 @@ def to_text(primitive=None, hexstr=None, text=None):
     raise TypeError("Expected an int, bytes, bytearray or hexstr.")
 
 
-def text_if_str(to_type, text_or_primitive):
+def text_if_str(to_type: Callable[..., T], text_or_primitive: Union[bytes, int, str]) -> T:
```

Comment:
> Basically, to_type will be something like Callable[[Optional[Primative], Optional[str], Optional[str]], Generic[T] where T = TypeVar('T'). Then hexstr_if_hex should be marked as returning type T. This may not be exactly right, but the idea is that we setup a generic type T which to_type returns and say that hexstr_if_hex will always return the same type as to_type.

This is what I got - Apparently `Generic` is just a Base class and not technically treated as a type. And from what I can tell, we need to use the `...` here instead of the `Optionals` b/c of the kwarg on line 130/1515 - a la https://github.com/python/mypy/issues/1655

Reply:
Honestly, do what you need to make this work and open an issue if you're not happy with it as a final solution.  It's possible that we can't really type these types of functions very well with the current state of mypy.